Framework/PythonInterface/mantid/api/src/Exports/WorkspaceGroup.cpp

```diff
@@ -90,7 +89,6 @@ void export_WorkspaceGroup() {
       .def("remove", &WorkspaceGroup::remove,
            (arg("self"), arg("workspace_name")), "Remove a name from the group")
       .def("getItem", getItem, (arg("self"), arg("index")),
-           return_value_policy<Policies::ToWeakPtr>(),
```

Comment:
Thanks for finding this whole in the setup.

I see what you're saying regarding the fixed type requirement in the policy as it needs to return the correct thing from `get_pytype()`. I think in this case we could use the local `getItem` intermediary that is already defined above. If we alter that to return a `boost::python::object` directly then we don't need the policy and we can decide in that function how to do the conversion. What do you think?

My main concern is interface code that holds onto one of these references that then means deleting something in the workspaces view will leave a workspacin memory with no way to delete it.

Reply:
Actually that's a really elegant way to fix it, thanks.
Looking back I think I was so busy trying to write a generic wrapper to decay to weak pointer. Coming back to it, as of this PR, this is the only place we decayed to a weak pointer. So taking the agile approach of not trying to anticipate future needs made it much easier.